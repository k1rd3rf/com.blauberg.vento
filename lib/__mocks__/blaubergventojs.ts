export * from 'blaubergventojs';

export const sendMock = jest.fn((packet, ip) =>
  Promise.resolve({ packet, ip })
);

export class BlaubergVentoClient {
  send = sendMock;

  findDevices = jest
    .fn()
    .mockResolvedValue([{ id: 'TEST1234', ip: '127.0.0.3' }]);

  findById = (id: string) => Promise.resolve({ id, ip: '127.0.0.2' });
}
