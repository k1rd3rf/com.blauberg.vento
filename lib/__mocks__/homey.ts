// eslint-disable-next-line max-classes-per-file
import { callsWithArgs, callCount } from '../testTools';

const setCapabilityValueMock = jest.fn();
const hasCapabilityMock = jest.fn().mockReturnValue(true);
const registerCapabilityListenerMock = jest.fn().mockReturnValue(true);
const registerRunListenerMock = jest.fn();
const getCapabilityValueMock = jest.fn();

const logMock = jest.fn();

const capabilities: Record<string, (value: unknown) => Promise<void>> = {};
const capabilityValue: Record<string, unknown> = {};

class FlowCard {
  capability: string;
  device: unknown;

  constructor(capability: string, device: unknown) {
    this.capability = capability;
    this.device = device;
  }

  registerRunListener = (fn: (value: unknown) => Promise<void>) => {
    capabilities[this.capability] = fn;
    registerRunListenerMock(this.capability, fn);
    return Promise.resolve({
      device: this.device,
      capability: this.capability,
      [this.capability]: capabilityValue[this.capability],
    });
  };

  trigger = (...args: object[]) => {
    logMock('triggered', this.capability, args);
    return Promise.resolve(true);
  };
}

// eslint-disable-next-line import/prefer-default-export
export class Device {
  homey = {
    flow: {
      getActionCard: (capability: string) => new FlowCard(capability, this),
      getConditionCard: (capability: string) => new FlowCard(capability, this),
      getDeviceTriggerCard: (capability: string) =>
        new FlowCard(capability, this),
    },
    setInterval: jest.fn(),
  };

  settings: Record<string, unknown> = {
    devicepwd: 'password',
  };

  store: Record<string, unknown> = {};

  log = (...args: unknown[]) => logMock('log', ...args);
  error = (...args: unknown[]) => logMock('error', ...args);

  getData() {
    return { id: 'TEST1234' };
  }

  async setCapabilityValue(capabilityId: string, value: unknown) {
    const fn = capabilities[capabilityId];
    capabilityValue[capabilityId] = value;
    if (fn) {
      const result = await fn({ value, device: this });
      return setCapabilityValueMock(capabilityId, value, { result });
    }
    return setCapabilityValueMock(
      capabilityId,
      value,
      'calling unregistered capability'
    );
  }

  getCapabilityValue(capabilityId: string) {
    getCapabilityValueMock(capabilityId);
    return capabilityValue[capabilityId];
  }

  async hasCapability(capabilityId: string) {
    hasCapabilityMock(capabilityId);
    return !!capabilities[capabilityId];
  }

  setStoreValue(key: string, value: unknown) {
    this.store[key] = value;
  }

  getStoreValue(key: string) {
    return this.store[key];
  }

  async registerCapabilityListener(
    rId: string,
    callback: (value: unknown) => Promise<void>
  ) {
    capabilities[rId] = callback;
    return registerCapabilityListenerMock(rId, callback);
  }

  getSetting(key: string) {
    return this.settings[key];
  }

  setSettings(settings: object) {
    // eslint-disable-next-line no-return-assign
    return (this.settings = { ...this.settings, ...settings });
  }

  async setUnavailable(reason?: string) {
    logMock('setUnavailable was called', reason);
  }

  async setAvailable(reason?: string) {
    logMock('setAvailable', reason);
  }

  getAvailable() {
    return true;
  }

  getMockCalls = () => ({
    setCapabilityValue: callsWithArgs(setCapabilityValueMock.mock.calls),
    hasCapability: callCount(hasCapabilityMock.mock.calls),
    registerCapabilityListener: callCount(
      registerCapabilityListenerMock.mock.calls
    ),
    registerRunListener: callsWithArgs(registerRunListenerMock.mock.calls),
    getCapabilityValue: callCount(getCapabilityValueMock.mock.calls),
    log: logMock.mock.calls,
    settings: this.settings,
    capabilities,
    store: this.store,
  });
}

export class Driver {
  homey = {
    setInterval: jest.fn(),
  };

  log = (...args: unknown[]) => logMock('log', ...args);
  error = (...args: unknown[]) => logMock('error', ...args);
}
