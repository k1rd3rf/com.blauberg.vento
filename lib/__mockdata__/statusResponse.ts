import { Response, Parameter } from 'blaubergventojs';

const onResponse = { parameter: 1, value: new Uint8Array([1]) };

const dataEntries = [
  { parameter: 2, value: new Uint8Array([1]) },
  { parameter: 68, value: new Uint8Array([179]) },
  { parameter: 6, value: new Uint8Array([0]) },
  { parameter: 102, value: new Uint8Array([30]) },
  { parameter: 183, value: new Uint8Array([2]) },
  { parameter: 136, value: new Uint8Array([0]) },
  { parameter: 100, value: new Uint8Array([34, 20, 31, 0]) },
  { parameter: 37, value: new Uint8Array([45]) },
  { parameter: 15, value: new Uint8Array([1]) },
  { parameter: 25, value: new Uint8Array([70]) },
  { parameter: 185, value: new Uint8Array([4, 0]) },
  { parameter: 74, value: new Uint8Array([236, 4]) },
  { parameter: 7, value: new Uint8Array([0]) },
  { parameter: 131, value: new Uint8Array([0]) },
  { parameter: 11 as Parameter, value: new Uint8Array([0, 0, 0]) },
];

export const statusResponse: Response = {
  // @ts-expect-error: mock data
  packet: {
    _deviceId: 'devId',
    _password: '1234',
    _functionType: 6,
    dataEntries: [onResponse, ...dataEntries],
  },
  ip: '127.0.0.1',
};

export const offResponse: Response = {
  // @ts-expect-error: mock data
  packet: {
    _deviceId: 'devId',
    _password: '1234',
    _functionType: 6,
    dataEntries: [{ parameter: 1, value: new Uint8Array([0]) }, ...dataEntries],
  },
  ip: '127.0.0.1',
};
