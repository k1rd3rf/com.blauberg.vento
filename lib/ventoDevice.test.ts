import VentoDevice from './ventoDevice';
import { Device } from './__mocks__/homey';
import { statusResponse, offResponse } from './__mockdata__/statusResponse';
import { sendMock } from './__mocks__/blaubergventojs';

jest.mock('./ventoDiscovery', () =>
  jest.fn().mockImplementation(() => ({
    findById: (id: string) => Promise.resolve({ id, ip: '127.0.0.2' }),
  }))
);

describe('ventoDevice', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  beforeEach(() => {
    sendMock.mockResolvedValue(statusResponse);
  });

  it('should be able to get status from modbus on init', async () => {
    const device = new VentoDevice();
    await device.onInit();

    expect({
      apiCalls: (device as unknown as Device).getMockCalls(),
    }).toMatchSnapshot();
  });
  it('should be able to get status when device is marked as off', async () => {
    const device = new VentoDevice();
    sendMock.mockResolvedValue(offResponse);
    await device.onInit();

    expect({
      apiCalls: (device as unknown as Device).getMockCalls(),
    }).toMatchSnapshot();
  });
  it('should be able to handle errors', async () => {
    const device = new VentoDevice();
    sendMock.mockRejectedValue(new Error('test errors'));

    await device.onInit();

    expect({
      apiCalls: (device as unknown as Device).getMockCalls(),
    }).toMatchSnapshot();
  });

  it('should get and set values trough the api and modbus', async () => {
    const device = new VentoDevice();
    await device.onInit();

    expect({
      modbusCalls: sendMock.mock.calls,
    }).toMatchSnapshot();
  });

  it('should init the api', async () => {
    const device = new VentoDevice();
    await device.onInit();

    const { deviceId, devicePass, deviceIp } = device.api;
    expect({ devicePass, deviceIp, deviceId }).toMatchSnapshot();
  });

  it('should be able to setup all capabilities', async () => {
    const device = new VentoDevice();
    await device.setupCapabilities();

    expect({
      calls: (device as unknown as Device).getMockCalls(),
    }).toMatchSnapshot();
  });
});
