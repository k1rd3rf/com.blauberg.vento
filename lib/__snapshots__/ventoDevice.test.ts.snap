// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ventoDevice should be able to get status from modbus on init 1`] = `
{
  "apiCalls": {
    "capabilities": {
      "alarm_boost": [Function],
      "alarm_filter": [Function],
      "alarm_generic": [Function],
      "fan_speed": [Function],
      "manualSpeed": [Function],
      "manualSpeed_set": [Function],
      "onoff": [Function],
      "operationMode": [Function],
      "operation_mode": [Function],
      "speedMode": [Function],
      "speed_mode": [Function],
      "timerMode": [Function],
      "timer_mode": [Function],
    },
    "getCapabilityValue": {
      "alarm_boost": 1,
      "alarm_filter": 1,
      "alarm_generic": 1,
    },
    "hasCapability": {
      "alarm_boost": 1,
      "alarm_connectivity": 1,
      "alarm_filter": 1,
      "alarm_generic": 1,
      "fan_speed": 2,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "log": [
      [
        "log",
        "Locating device with id TEST1234",
      ],
      [
        "log",
        "Create the flow for the speed mode capability",
      ],
      [
        "log",
        "Create the flow for the manual speed capability",
      ],
      [
        "log",
        "Create the flow for the operation mode capability",
      ],
      [
        "log",
        "Create the flow for the timer mode capability",
      ],
      [
        "log",
        "Requesting the current device state",
      ],
      [
        "log",
        "Device state received: ",
        {
          "alarm": 0,
          "boost": {
            "deactivationtimer": 30,
            "mode": 0,
          },
          "fan": {
            "rpm": 236,
          },
          "filter": {
            "alarm": 0,
            "timer": {
              "days": 31,
              "hour": 20,
              "min": 34,
            },
          },
          "humidity": {
            "activated": 0,
            "current": 45,
            "sensoractivation": 1,
            "threshold": 70,
          },
          "onoff": 1,
          "operationmode": 2,
          "speed": {
            "manualspeed": 179,
            "mode": 1,
          },
          "timers": {
            "countdown": {
              "hour": 0,
              "min": 0,
              "sec": 0,
            },
            "mode": 0,
          },
          "unittype": "Vento Expert Duo A30-1 W V.2",
        },
      ],
      [
        "setAvailable",
        undefined,
      ],
    ],
    "registerCapabilityListener": {
      "fan_speed": 1,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "registerRunListener": {
      "alarm_boost": [
        [
          [Function],
        ],
      ],
      "alarm_filter": [
        [
          [Function],
        ],
      ],
      "alarm_generic": [
        [
          [Function],
        ],
      ],
      "manualSpeed_set": [
        [
          [Function],
        ],
      ],
      "operation_mode": [
        [
          [Function],
        ],
      ],
      "speed_mode": [
        [
          [Function],
        ],
      ],
      "timer_mode": [
        [
          [Function],
        ],
      ],
    },
    "setCapabilityValue": {
      "alarm_boost": [
        [
          false,
          {
            "result": false,
          },
        ],
      ],
      "alarm_connectivity": [
        [
          false,
          "calling unregistered capability",
        ],
      ],
      "alarm_filter": [
        [
          false,
          {
            "result": false,
          },
        ],
      ],
      "alarm_generic": [
        [
          false,
          {
            "result": false,
          },
        ],
      ],
      "fan_speed": [
        [
          0.7019607843137254,
          {
            "result": undefined,
          },
        ],
      ],
      "filter_timer": [
        [
          "31:20:34",
          "calling unregistered capability",
        ],
      ],
      "manualSpeed": [
        [
          70.19607843137254,
          {
            "result": undefined,
          },
        ],
      ],
      "measure_RPM": [
        [
          236,
          "calling unregistered capability",
        ],
      ],
      "measure_humidity": [
        [
          45,
          "calling unregistered capability",
        ],
      ],
      "onoff": [
        [
          true,
          {
            "result": undefined,
          },
        ],
      ],
      "operationMode": [
        [
          "2",
          {
            "result": undefined,
          },
        ],
      ],
      "speedMode": [
        [
          "1",
          {
            "result": undefined,
          },
        ],
      ],
      "timerMode": [
        [
          "0",
          {
            "result": undefined,
          },
        ],
      ],
      "timerMode_timer": [
        [
          "0:0:0",
          "calling unregistered capability",
        ],
      ],
    },
    "settings": {
      "boost_delay": 30,
      "devicemodel": "Vento Expert Duo A30-1 W V.2",
      "devicepwd": "password",
      "humidity_sensor": true,
      "humidity_threshold": 70,
    },
  },
}
`;

exports[`ventoDevice should be able to get status when device is marked as off 1`] = `
{
  "apiCalls": {
    "capabilities": {
      "alarm_boost": [Function],
      "alarm_filter": [Function],
      "alarm_generic": [Function],
      "fan_speed": [Function],
      "manualSpeed": [Function],
      "manualSpeed_set": [Function],
      "onoff": [Function],
      "operationMode": [Function],
      "operation_mode": [Function],
      "speedMode": [Function],
      "speed_mode": [Function],
      "timerMode": [Function],
      "timer_mode": [Function],
    },
    "getCapabilityValue": {
      "alarm_boost": 1,
      "alarm_filter": 1,
      "alarm_generic": 1,
    },
    "hasCapability": {
      "alarm_boost": 1,
      "alarm_connectivity": 1,
      "alarm_filter": 1,
      "alarm_generic": 1,
      "fan_speed": 2,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "log": [
      [
        "log",
        "Locating device with id TEST1234",
      ],
      [
        "log",
        "Create the flow for the speed mode capability",
      ],
      [
        "log",
        "Create the flow for the manual speed capability",
      ],
      [
        "log",
        "Create the flow for the operation mode capability",
      ],
      [
        "log",
        "Create the flow for the timer mode capability",
      ],
      [
        "log",
        "Requesting the current device state",
      ],
      [
        "log",
        "Device state received: ",
        {
          "alarm": 0,
          "boost": {
            "deactivationtimer": 30,
            "mode": 0,
          },
          "fan": {
            "rpm": 236,
          },
          "filter": {
            "alarm": 0,
            "timer": {
              "days": 31,
              "hour": 20,
              "min": 34,
            },
          },
          "humidity": {
            "activated": 0,
            "current": 45,
            "sensoractivation": 1,
            "threshold": 70,
          },
          "onoff": 0,
          "operationmode": 2,
          "speed": {
            "manualspeed": 179,
            "mode": 1,
          },
          "timers": {
            "countdown": {
              "hour": 0,
              "min": 0,
              "sec": 0,
            },
            "mode": 0,
          },
          "unittype": "Vento Expert Duo A30-1 W V.2",
        },
      ],
      [
        "setAvailable",
        undefined,
      ],
    ],
    "registerCapabilityListener": {
      "fan_speed": 1,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "registerRunListener": {
      "alarm_boost": [
        [
          [Function],
        ],
      ],
      "alarm_filter": [
        [
          [Function],
        ],
      ],
      "alarm_generic": [
        [
          [Function],
        ],
      ],
      "manualSpeed_set": [
        [
          [Function],
        ],
      ],
      "operation_mode": [
        [
          [Function],
        ],
      ],
      "speed_mode": [
        [
          [Function],
        ],
      ],
      "timer_mode": [
        [
          [Function],
        ],
      ],
    },
    "setCapabilityValue": {
      "alarm_boost": [
        [
          false,
          {
            "result": false,
          },
        ],
      ],
      "alarm_connectivity": [
        [
          false,
          "calling unregistered capability",
        ],
      ],
      "alarm_filter": [
        [
          false,
          {
            "result": false,
          },
        ],
      ],
      "alarm_generic": [
        [
          false,
          {
            "result": false,
          },
        ],
      ],
      "fan_speed": [
        [
          0.7019607843137254,
          {
            "result": undefined,
          },
        ],
      ],
      "filter_timer": [
        [
          "31:20:34",
          "calling unregistered capability",
        ],
      ],
      "manualSpeed": [
        [
          70.19607843137254,
          {
            "result": undefined,
          },
        ],
      ],
      "measure_RPM": [
        [
          236,
          "calling unregistered capability",
        ],
      ],
      "measure_humidity": [
        [
          45,
          "calling unregistered capability",
        ],
      ],
      "onoff": [
        [
          false,
          {
            "result": undefined,
          },
        ],
      ],
      "operationMode": [
        [
          "2",
          {
            "result": undefined,
          },
        ],
      ],
      "speedMode": [
        [
          "1",
          {
            "result": undefined,
          },
        ],
      ],
      "timerMode": [
        [
          "0",
          {
            "result": undefined,
          },
        ],
      ],
      "timerMode_timer": [
        [
          "0:0:0",
          "calling unregistered capability",
        ],
      ],
    },
    "settings": {
      "boost_delay": 30,
      "devicemodel": "Vento Expert Duo A30-1 W V.2",
      "devicepwd": "password",
      "humidity_sensor": true,
      "humidity_threshold": 70,
    },
  },
}
`;

exports[`ventoDevice should be able to handle errors 1`] = `
{
  "apiCalls": {
    "capabilities": {
      "alarm_boost": [Function],
      "alarm_filter": [Function],
      "alarm_generic": [Function],
      "fan_speed": [Function],
      "manualSpeed": [Function],
      "manualSpeed_set": [Function],
      "onoff": [Function],
      "operationMode": [Function],
      "operation_mode": [Function],
      "speedMode": [Function],
      "speed_mode": [Function],
      "timerMode": [Function],
      "timer_mode": [Function],
    },
    "getCapabilityValue": {},
    "hasCapability": {
      "alarm_boost": 1,
      "alarm_connectivity": 1,
      "alarm_filter": 1,
      "alarm_generic": 1,
      "fan_speed": 2,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "log": [
      [
        "log",
        "Locating device with id TEST1234",
      ],
      [
        "log",
        "Create the flow for the speed mode capability",
      ],
      [
        "log",
        "Create the flow for the manual speed capability",
      ],
      [
        "log",
        "Create the flow for the operation mode capability",
      ],
      [
        "log",
        "Create the flow for the timer mode capability",
      ],
      [
        "log",
        "Requesting the current device state",
      ],
      [
        "setUnavailable was called",
        undefined,
      ],
      [
        "error",
        "Failed to get device state, device unreachable",
        [Error: test errors],
      ],
    ],
    "registerCapabilityListener": {
      "fan_speed": 1,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "registerRunListener": {
      "alarm_boost": [
        [
          [Function],
        ],
      ],
      "alarm_filter": [
        [
          [Function],
        ],
      ],
      "alarm_generic": [
        [
          [Function],
        ],
      ],
      "manualSpeed_set": [
        [
          [Function],
        ],
      ],
      "operation_mode": [
        [
          [Function],
        ],
      ],
      "speed_mode": [
        [
          [Function],
        ],
      ],
      "timer_mode": [
        [
          [Function],
        ],
      ],
    },
    "setCapabilityValue": {
      "alarm_connectivity": [
        [
          true,
          "calling unregistered capability",
        ],
      ],
    },
    "settings": {
      "devicepwd": "password",
    },
  },
}
`;

exports[`ventoDevice should be able to setup all capabilities 1`] = `
{
  "calls": {
    "capabilities": {
      "alarm_boost": [Function],
      "alarm_filter": [Function],
      "alarm_generic": [Function],
      "fan_speed": [Function],
      "manualSpeed": [Function],
      "manualSpeed_set": [Function],
      "onoff": [Function],
      "operationMode": [Function],
      "operation_mode": [Function],
      "speedMode": [Function],
      "speed_mode": [Function],
      "timerMode": [Function],
      "timer_mode": [Function],
    },
    "getCapabilityValue": {},
    "hasCapability": {
      "alarm_boost": 1,
      "alarm_filter": 1,
      "alarm_generic": 1,
      "fan_speed": 1,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "log": [
      [
        "log",
        "Create the flow for the speed mode capability",
      ],
      [
        "log",
        "Create the flow for the manual speed capability",
      ],
      [
        "log",
        "Create the flow for the operation mode capability",
      ],
      [
        "log",
        "Create the flow for the timer mode capability",
      ],
    ],
    "registerCapabilityListener": {
      "fan_speed": 1,
      "manualSpeed": 1,
      "onoff": 1,
      "operationMode": 1,
      "speedMode": 1,
      "timerMode": 1,
    },
    "registerRunListener": {
      "alarm_boost": [
        [
          [Function],
        ],
      ],
      "alarm_filter": [
        [
          [Function],
        ],
      ],
      "alarm_generic": [
        [
          [Function],
        ],
      ],
      "manualSpeed_set": [
        [
          [Function],
        ],
      ],
      "operation_mode": [
        [
          [Function],
        ],
      ],
      "speed_mode": [
        [
          [Function],
        ],
      ],
      "timer_mode": [
        [
          [Function],
        ],
      ],
    },
    "setCapabilityValue": {},
    "settings": {
      "devicepwd": "password",
    },
  },
}
`;

exports[`ventoDevice should get and set values trough the api and modbus 1`] = `
{
  "modbusCalls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": undefined,
          },
          {
            "parameter": 2,
            "value": undefined,
          },
          {
            "parameter": 68,
            "value": undefined,
          },
          {
            "parameter": 6,
            "value": undefined,
          },
          {
            "parameter": 102,
            "value": undefined,
          },
          {
            "parameter": 183,
            "value": undefined,
          },
          {
            "parameter": 136,
            "value": undefined,
          },
          {
            "parameter": 100,
            "value": undefined,
          },
          {
            "parameter": 37,
            "value": undefined,
          },
          {
            "parameter": 15,
            "value": undefined,
          },
          {
            "parameter": 25,
            "value": undefined,
          },
          {
            "parameter": 185,
            "value": undefined,
          },
          {
            "parameter": 74,
            "value": undefined,
          },
          {
            "parameter": 7,
            "value": undefined,
          },
          {
            "parameter": 131,
            "value": undefined,
          },
          {
            "parameter": 11,
            "value": undefined,
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 1,
        "_password": "password",
      },
      "127.0.0.2",
    ],
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.2",
    ],
    [
      E {
        "_dataEntries": [
          {
            "parameter": 2,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.2",
    ],
    [
      E {
        "_dataEntries": [
          {
            "parameter": 68,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.2",
    ],
    [
      E {
        "_dataEntries": [
          {
            "parameter": 68,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.2",
    ],
    [
      E {
        "_dataEntries": [
          {
            "parameter": 183,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.2",
    ],
    [
      E {
        "_dataEntries": [
          {
            "parameter": 7,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "TEST1234",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.2",
    ],
  ],
}
`;

exports[`ventoDevice should init the api 1`] = `
{
  "deviceId": "TEST1234",
  "deviceIp": "127.0.0.2",
  "devicePass": "password",
}
`;
